aspnetcore/fundamentals/servers/yarp/ab-testing.md

```diff
@@ -0,0 +1,58 @@
+---
+uid: fundamentals/servers/yarp/ab-testing
+title: A/B Testing and Rolling Upgrades
+description: A/B Testing and Rolling Upgrades
+author: rick-anderson
+ms.author: riande
+ms.date: 02/06/2025
+ms.topic: article
+content_well_notification: AI-contribution
+ms.prod: aspnet-core
+ai-usage: ai-assisted
+---
+
+# A/B Testing and Rolling Upgrades
+
+## Introduction
+
+A/B testing and rolling upgrades require procedures for dynamically assigning incoming traffic to evaluate changes in the destination application. YARP does not have a built-in model for this, but it does expose some infrastructure useful for building such a system. See [issue #126](https://github.com/microsoft/reverse-proxy/issues/126) for additional details about this scenario.
+
+## Example
+
+```
```

Comment:
Should these code blocks have ```csharp so the syntax highlighting is correct?

Reply:
yes, next pass. This pass is to get a clean build. The PR merges into a staging branch.

Reply:
@adityamandaleeka 
> Should these code blocks have ```csharp so the syntax highlighting is correct?

Ultimately, we'll move all the code snippets to project files and import the code. That way we can make sure they compile.  That allows highlights too.